ARCs/arc-0027.md

```diff
@@ -0,0 +1,1611 @@
+---
+arc: 27
+title: Web Extension Wallet Interface
+description: A common interface for multiple web extension providers to interact with dapps.
+author: Kieran O'Neill (@kieranroneill)
+status: Draft
+type: Standards Track
+category: Interface
+created: 2023-12-24
+---
+
+# Web Extension Provider Interface
+
+## Abstract
+
+Building off of the previous ARCs relating to; wallet transaction signing ([ARC-0005](./arc-0005.md#specification)), wallet address discovery ([ARC-0006](./arc-0006.md#specification)), wallet transaction network posting ([ARC-0007](./arc-0007.md#specification)) and wallet transaction signing & posting ([ARC-0008](./arc-0008.md#specification)), this proposal aims to extend and comprehensively outline a common interface between multiple web extension wallets and dapps by utilizing the pub/sub semantics of the <a href="https://developer.mozilla.org/en-US/docs/Web/API/Broadcast_Channel_API">BroadcastChannel API</a>.
```

Comment:
Using a  `BroadcastChannel` is an interesting choice and would need a bit more inspection. AFAIK web extensions do not typically inject in iframes, with this approach we may be allowing embedded iframes to trigger a wallet popup as long as that origin is also open in a tab elsewhere.

Reply:
Do I understand it correctly that if multiple tabs are open and thus content scripts are listening to the BroadcastChannel, then they need to avoid registering multiple times? If registering multiple times is allowed, would the content script need to synchronize internally to ensure it only registers once and doesn't deal with the request multiple times?